### DISCUSSION TITLE
Cannot connect SQL Server with DBeaver 21

### DISCUSSION THREAD

**Discussion by penCsharpener** [dbeaver/dbeaver#11865] [Q&A]
Although there are various similar issues about this around I found none that helped me so far - neither at github nor stackoverflow or blogs...

#### System information: 
- Operating system (distribution) and version: Windows 10 20H2
- DBeaver version: 21
- Additional extensions: just plain DBeaver

#### Connection specification:
- Database name and version: SQL Server 2019 Developer Edition installed directly on Windows 10
- Driver name: not sure - mssql-jdbc:RELEASE 8.2.0.jre8 ?
- Do you use tunnels or proxies (SSH, SOCKS, etc)? nope, just trying localhost connection

#### Describe the problem you're observing:
I'm trying to connect to my local instance and it just times out all the time. I already checked that the login users are Enabled, Grants are giving and the password is correct. It works in my dotnet applications without problems and also in SSMS and Azure Data Studio.

#### Steps to reproduce, if exist:
not sure. This is the simplest setup I can think of. 

#### Include any warning/errors/backtraces from the logs
The TCP/IP connection to the host localhost, port 1433 has failed. Error: "Connection timed out: no further information.. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.".

**Comment by penCsharpener** [comment]
Here is my config

![image](https://user-images.githubusercontent.com/26190934/110915169-960f3e80-8317-11eb-9fe0-d51dc67209c7.png)

![image](https://user-images.githubusercontent.com/26190934/110915277-b808c100-8317-11eb-98ea-a3cd3df1453d.png)

![image](https://user-images.githubusercontent.com/26190934/110915366-d373cc00-8317-11eb-8290-820ae9c670c3.png)
![image](https://user-images.githubusercontent.com/26190934/110915391-dc649d80-8317-11eb-8918-81ae3952badf.png)

**Comment by kseniiaguzeeva** [comment]
Please try to write 0.0.0.0 instead of localhost, check whether the port is correct for your local SQL Server.
![dbeaver_RG5UxMhWJD](https://user-images.githubusercontent.com/51405061/110928912-72a8bb80-8338-11eb-9638-508858288a19.png)

**Comment by uslss** [comment]
@penCsharpener also please check tcp/ip settings for sql server instance
https://kb.sos-berlin.com/pages/viewpage.action?pageId=17499564
https://stackoverflow.com/questions/18841744/jdbc-connection-failed-error-tcp-ip-connection-to-host-failed

**Comment by penCsharpener** [comment]
I tried all the above. No luck. My TCP port is 1433 and that works fine in Azure Data Studio and SSMS. 
But when I run `telnet 127.0.0.1 1433` it also says 

```
Connecting To 127.0.0.1...Could not open connection to the host, on port 1433: Connect failed
```

I mean this is a fresh installation with all default settings. I can connect DBeaver to a remove SQL server but for some reason not to my localhost instance. It even worked in the past with a docker container version but I had to give up on that because the performance was much less then the regular installation of SQL Server. 


![image](https://user-images.githubusercontent.com/26190934/110983428-05ae1980-836a-11eb-834c-6ac0e58c9ae7.png)

**Comment by uslss** [comment]
@penCsharpener could you check if tcp/ip is enabled for your mssql instance?
https://docs.microsoft.com/en-us/sql/database-engine/configure-windows/configure-a-server-to-listen-on-a-specific-tcp-port?view=sql-server-ver15

**Comment by penCsharpener** [comment]
It says Active yes, and Enable No but even if I do Enabled Yes and Active Yes for TCP/IP and restart the service I can't connect.

**Comment by uslss** [comment]
SMSS and Azure Data Studio don't use TCP/IP protocol to connect to MSSQL, so they will connect successfully even if the TCP/IP configuration is wrong.
Please recheck your Database configuration.

**Comment by penCsharpener** [comment]
Finally I got it. 
This helped me: https://support.ipconfigure.com/hc/en-us/articles/201661259-TCP-IP-Settings-in-SQL-Configuration-Manager
point 5. in the list
